src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -253,7 +254,7 @@ public void AddProject(PackageSpec projectSpec)
                     projectToRestore = ToPackageSpecWithCentralVersionInformation(projectSpec);
                 }
 
-                _projects.Add(projectUniqueName, projectToRestore);
+                _projects.TryAdd(projectUniqueName, projectToRestore);
```

Comment:
The semantics here are probably changing. I'm not sure about `SortedDictionary<TKey, TValue>`, but `Dictionary<TKey, TValue>` throws when you try to call `.Add` on a key that already exists. The new code will silently ignore the new value.